modules/javafx.graphics/src/main/java/com/sun/prism/es2/ES2Light.java

```diff
@@ -58,6 +61,10 @@
     }
 
     boolean isPointLight() {
-        return falloff == 0 && outerAngle == 180;
+        return falloff == 0 && outerAngle == 180 && isAttenuated > 0.5;
+    }
+
+    boolean isDirectionalLight() {
+        return isAttenuated < 0.5;
```

Comment:
Minor: you can just test for 0 or not 0 since you only ever set them as constants. It's up to you...I don't object to rounding, which is effectively what you've done.